lib/stack-cache-generator.nix

```diff
@@ -69,7 +69,6 @@ concatMap (dep:
                 }
                 else builtins.fetchGit {
                   inherit (dep) url rev;
-                  ref = "*";
                 };
```

Comment:
Do you remember why this was here?  When I was updating `cabalProject` recently to work without a `sha256` I found that passing the `tag` field from the cabal file as `ref` seemed to work with a hash or a branch name (the tag can be either).  Is that the right thing to do?

https://github.com/input-output-hk/haskell.nix/blob/master/lib/call-cabal-project-to-nix.nix#L122

Reply:
The difference between `ref = "*";` and no `ref` is that in the first case `fetchGit` fetches the entire repo and then checks out `rev`, and in the second case it only fetches the default branch. Former functionality was deemed to be a bug rather than a feature and removed in nix 2.4pre (and you need 2.4pre to get flakes).

Yes, what you're doing for cabal files is likely correct. However, I don't know of such an attribute in `stack.yaml` (if there is one, it appears to be undocumented).